assignment/impl/src/java/org/sakaiproject/assignment/impl/reminder/AssignmentDueReminderServiceImpl.java

```diff
@@ -139,10 +141,31 @@ public void execute(String assignmentId) {
         try {
             Assignment assignment = assignmentService.getAssignment(assignmentId);
             Site site = siteService.getSite(assignment.getContext());
-
+            
+            Set<AssignmentSubmission> submissions = assignmentService.getSubmissions(assignment);
+            // Get submitters
+            List<String> submitters = submissions.stream()
+                    .filter(s -> s.getUserSubmission() == true)
+                    .flatMap(s -> s.getSubmitters().stream())
+                    .map(AssignmentSubmissionSubmitter::getSubmitter)
+                    .collect(Collectors.toList());
+            
+            // Get all members in a List collection
+            List<Member> membersList = site.getMembers().stream().collect(Collectors.toList());
+            // List with members filtered (members who are not submitters)
+            List<Member> membersNotSubmitters = new ArrayList<>();
+            
+            for (int i = 0; i < membersList.size(); i++) {
+                for (int j = 0; j < submitters.size(); j++) {
+                    if(!membersList.get(i).getUserId().equals(submitters.get(j))) {
+                        membersNotSubmitters.add(membersList.get(i));
+                    }
+                }
+            }
+            
             // Do not send reminders if the site is unpublished or softly deleted
             if (site.isPublished() && !site.isSoftlyDeleted()) {
-                for (Member member : site.getMembers()) {
+                for (Member member : membersNotSubmitters) {
                     if (member.isActive() && assignmentService.canSubmit(assignment, member.getUserId()) && !assignmentService.allowAddAssignment(assignment.getContext(), member.getUserId()) && checkEmailPreference(member)) {
```

Comment:
canSubmit is just a permission check or checks the submission?

Reply:
canSubmit is the magic method that should do filtering:

1) does the user have permission
2) does the user have ability to submit a new submission (e.g., has not used all their submissions)
3) if group submission, is a member of the proper group

Please see old JIRAs like:

* SAK-43674 Assignment reminder emails will never send (#8236)
* SAK-42667 Assignments: don't send reminder to instructors (#7486)
* SAK-42380 Assignment canSubmit will properly check permissions (#7401)
* SAK-41865 only send mail to an active member (#6972)
* SAK-40544 add 24 hour reminder email to students for assignments (#5942)

Reply:
Thanks for the detailed info @ottenhoff, you resolved all the investigation I told @druiz309 to do hehe 💯 

A 20.x client complains about this but I believe this problem does not happen in 20.x anymore, @druiz309 is investigating the issue.

Thanks again